ShopApp/ShopApp/Controller/HomeViewController.swift

```diff
@@ -33,52 +41,66 @@ class HomeViewController: UIViewController {
         navigationController?.setViewControllers([loginViewController], animated: true)
     }
 
-    func addUser() async {
-        let ref = Database.database().reference()
-
-        guard let name = self.currentUser?.displayName as? String else { return }
-        guard let email = self.currentUser?.email as? String else { return }
-
-        do {
-            // Get data from users table
-            let dataRead = await ref.child("users").observeSingleEventAndPreviousSiblingKey(of: .value)
-
-            let data = dataRead.0.value as? [String: Any]
-
-            // Check if data is null
-            guard let data = data else {
-                return
-            }
-
-            let emailVal = email.replacingOccurrences(of: ".", with: " ")
-
-            // For every user in users table
-            for dataKeys in data.keys where dataKeys.elementsEqual(emailVal) {
-                return
-            }
-
-            let refValue = Database.database().reference().child("users/\(emailVal)")
-            try await refValue.updateChildValues(["name": name])
-        } catch {
-            print(error)
-        }
-    }
-
+    // MARK: - ViewDidLoad
     override func viewDidLoad() {
         super.viewDidLoad()
-        async { await self.addUser() }
-        self.configureCategoryCollectionView()
-    }
 
-    func configureCategoryCollectionView() {
         let cellIdentifier = "CategoryCollectionViewCell"
         let layout = UICollectionViewFlowLayout()
         layout.itemSize = CGSize(width: UIScreen.main.bounds.width / 3, height: 120)
         collectionView.collectionViewLayout = layout
 
-        collectionView.register(CategoryCollectionViewCell.nib(), forCellWithReuseIdentifier: cellIdentifier)
-        collectionView.delegate = self
-        collectionView.dataSource = self
+        self.collectionView.register(CategoryCollectionViewCell.nib(), forCellWithReuseIdentifier: cellIdentifier)
+        self.collectionView.delegate = self
+        self.collectionView.dataSource = self
+
+        self.getAllCategories()
+    }
+
+    // MARK: - Get all categories from database
+    func getAllCategories() {
+        // Read data from database/categories/
+        ref.child("categories/").observeSingleEvent(of: .value, with: { snapshot in
+            // Get categories and images as dictionary
+            let value = snapshot.value as? NSDictionary
+            // Unwrap the dictionary
+            if let categories = value {
+                // For each category element get the attributes
+                for category in categories {
+                    var newCategory = Category()
+                    // Set the name of category
+                    newCategory.setName(name: category.key as! String)
+                    // Create the path to retrieve the image
+                    let path = category.value as! String
```

Comment:
instead of force unwrapping you could cast the snapshot.value to [String: String] instead of NSDictionary